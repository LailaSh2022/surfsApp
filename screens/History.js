import React from "react";
import { View, Text, StyleSheet } from "react-native";
import HistoryItem from "./../components/HistoryItem";
import { StyledContainer } from "./../components/Styles";
import { GetAllOrderByUserId } from "../Database";

const History = () => {
  let transactions = [
    {
      OrderNo: "2458",
      SentDate: "Jan 18 2023",
      Receiver: "Tatiane Rodrigues",
      Amount: "1500NZD",
      ReceivierGets: "5000BRL",
    },
    {
      OrderNo: "2459",
      SentDate: "Jan 19 2023",
      Receiver: "Laila Shihada",
      Amount: "1600NZD",
      ReceivierGets: "6000BRL",
    },
    {
      OrderNo: "2458",
      SentDate: "Jan 20 2023",
      Receiver: "Sophon Keo",
      Amount: "1700NZD",
      ReceivierGets: "7000BRL",
    },
  ];

  GetAllOrderByUserId(1)
    .then((result) => {
      console.log(result);
      const orders = result;
      orders.map((order) => console.log(order));

      /*
      GetReceiverBankInfo(receiver.Bank_Info)
        .then((result) => {
          bank_info = result;
        })
        .catch((error) => {
          console.log(`Error while getting bank_info: ${error}`);
          return;
        });
        */
    })
    .catch((error) => {
      console.log(`Error while getting order details: ${error}`);
      return;
    });

  return (
    <StyledContainer>
      <View>
        <View style={{ flexDirection: "row" }}>
          <Text style={styles.headline}> History</Text>
        </View>
        <Text style={{ paddingTop: 15, color: "blue", fontSize: 15 }}>
          Transactions
        </Text>
        {transactions.map((item, i) => (
          <HistoryItem item={item} key={i} />
        ))}
      </View>
    </StyledContainer>
  );
};

const styles = StyleSheet.create({
  headline: {
    flex: 1,
    textAlign: "center",
    fontWeight: "bold",
    fontSize: 18,
    paddingTop: 10,
    color: "#152A5F",
  },
});

export default History;
