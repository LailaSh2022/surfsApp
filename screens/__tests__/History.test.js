import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import History from "../History";

jest.mock("@react-navigation/native", () => ({
  useRoute: () => ({
    params: {
      userId: "3",
    },
  }),
}));

jest.mock("../../Database", () => ({
  GetAllOrderByUserId: jest.fn(() => Promise.resolve([])),
}));

it(
  "Testing History screen against its snapshot. " +
    "If there is any unexpected change in the screen, it will prompt test failed.",
  () => {
    const tree = renderer.create(<History />).toJSON();
    expect(tree).toMatchSnapshot();
  }
);
