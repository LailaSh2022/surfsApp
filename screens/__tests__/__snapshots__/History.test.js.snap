// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing History screen against its snapshot. If there is any unexpected change in the screen, it will prompt test failed. 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 25,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 64,
      },
    ]
  }
>
  <View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#152A5F",
            "borderRadius": 5,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#ffffff",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
        >
          &lt;
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#152A5F",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": "bold",
            "paddingTop": 10,
            "textAlign": "center",
          }
        }
      >
         History
      </Text>
    </View>
    <Text
      style={
        {
          "color": "blue",
          "fontSize": 15,
          "paddingTop": 15,
        }
      }
    >
      Transactions
    </Text>
  </View>
</View>
`;
